fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/exception/UnsupportedAlgorithmException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.exception;
+
+/**
+ * @author harring
+ * @since 2017-09-12
+ */
+public class UnsupportedAlgorithmException extends Exception {
```

Comment:
Would it be possible to just use the built-in `java.security.NoSuchAlgorithmException` here?  It's used "when a particular cryptographic algorithm is requested but is not available in the environment" -- do the semantics of `UnsupportedAlgorithmException` diverge much from that?

Reply:
Hmm, I guess I'm not sure whether we'd want a security-related exception to be thrown here. Are we thinking of checksums as a security feature here or as more related to fixity?

Reply:
Ah, I see.  It's because it is being used in a very specific circumstance, which demands a very specific response.  `NoSuchAlgorithmException` may be too general - i.e if it's thrown while doing a non-fixity-related operation, we might want a different response.

Reply:
@harringj  Would you be able to update the Javadoc for this class, with something like 

<pre>
/** Client uses an unsupported hash algorithm 
* &lt;p&gt;
* Thrown in circumstances where a client has used an unknown or unsupported algorithm in a request,
* e.g. with `Digest` or `Want-Digest`.
* &lt;/p&gt;
* @author harring 
* @since 2017-09-12  
*/ 
public class UnsupportedAlgorithmException extends Exception { 
</pre>

Reply:
Oh, my bad for overlooking the javadoc on this class!